File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        
        for(int i = 0; i<flowerbed.length; i++){
            if(i==0){
                if(flowerbed[i]==0 && (flowerbed.length==1 || flowerbed[i+1]==0)){
                    System.out.println("wjrldy");
                    n--;
                    flowerbed[i]=1;
                }
            }
            else if(i==flowerbed.length-1){
                if(flowerbed[i]==0 && flowerbed[i-1]==0){
                    n--;
                    flowerbed[i]=1;
                }
            }
            else if(flowerbed[i-1] == 0 && flowerbed[i] == 0 && flowerbed[i+1] == 0){
                n--;
                flowerbed[i]=1;
            }
        }
        System.out.println(n);
        if(n<=0)
            return true;
        else
            return false;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        for i in range(len(flowerbed)):
            if flowerbed[i] == 0 and (i-1 < 0 or flowerbed[i-1]==0) and (i+1 == len(flowerbed) or flowerbed[i+1]==0):
                flowerbed[i] = 1
                n-=1

        print(flowerbed)
        
        return n <= 0